### ISSUE TITLE
Add a section to the documentation site to document plugin-specific details

### ISSUE THREAD

**Issue by ddcprg** [apache/pinot#8775]
Not sure if this make sense but some plugins has particular features that should be documented somewhere in the docs rather than user having to search through the source code to find out about them. Som examples include configuration details for the Kinesis and Kafka stream ingestion, AVRO logical types switch, etc.

**Comment by KKcorps** [comment]
Yes, this should definitely be done. A good place would be to create a page for plugins in `CONFIGURATION REFERENCE` section in the docs.
